neo/rawio/neuronexusrawio.py

```diff
@@ -0,0 +1,275 @@
+        >>> raw_chunk = reader.get_analogsignal_chunk(block_index=0
+                seg_index=0,
+                stream_index=0)
+        # this isn't necessary for this reader since data is stored as float uV, but
+        # this is included in case there is a future change to the format
+        >>> float_chunk = reader.rescale_signal_raw_to_float(raw_chunk, stream_index=0)
+
+        """
+
+        BaseRawIO.__init__(self)
+
+        if not Path(filename).is_file():
+            raise FileNotFoundError(f"The metadata file {filename} was not found")
+        if Path(filename).suffix != ".json":
+            raise NeoReadWriteError(
+                f"The json metadata file should be given, filename entered is {Path(filename).stem}"
+            )
+        meta_filename = Path(filename)
+        binary_file = meta_filename.parent / (meta_filename.stem.split(".")[0] + "_data.xdat")
+
+        if not binary_file.exists() and not binary_file.is_file():
+            raise FileNotFoundError(f"The data.xdat file {binary_file} was not found. Is it in the same directory?")
+        timestamp_file = meta_filename.parent / (meta_filename.stem.split(".")[0] + "_timestamp.xdat")
+        if not timestamp_file.exists() and not timestamp_file.is_file():
+            raise FileNotFoundError(
+                f"The timestamps.xdat file {timestamp_file} was not found. Is it in the same directory?"
+            )
+
+        self.filename = filename
+        self.binary_file = binary_file
+        self.timestamp_file = timestamp_file
+
+    def _source_name(self):
+        # return the metadata filename only
+        return self.filename
+
+    def _parse_header(self):
+
+        # read metadata
+        self.metadata = self.read_metadata(self.filename)
+
+        # Collect information necessary for memory map
+        self._sampling_frequency = self.metadata["status"]["samp_freq"]
+        self._n_samples, self._n_channels = self.metadata["status"]["shape"]
+        # Stored as a simple float32 binary file
+        BINARY_DTYPE = "float32"
+        binary_file = self.binary_file
+        timestamp_file = self.timestamp_file
+
+        # Make the two memory maps
+        self._raw_data = np.memmap(
+            binary_file,
+            dtype=BINARY_DTYPE,
+            mode="r",
+            shape=(self._n_samples, self._n_channels),
+            offset=0,  # headerless binary file
+        )
+        self._timestamps = np.memmap(
+            timestamp_file,
+            dtype=np.int64,  # this is from the allego sample reader timestamps are np.int64
+            mode="r",
+            offset=0,  # headerless binary file
+        )
+
+        # We can do a quick timestamp check to make sure it is the correct timestamp data for the
+        # given metadata
+        if self._timestamps[0] != self.metadata["status"]["timestamp_range"][0]:
+            raise NeoReadWriteError(f"The metadata indicates a different starting timestamp")
```

Comment:
It would be good to show which ones.

Reply:
Do you want to do an actual suggestion? You want me to compare the timestamp file from the metadata file in the actual error?

Reply:
I mean showing/displaying the two timestamps in the error assertion.

Reply:
That's what I thought you meant but wanted to confirm. Yep I will add that.